File: http patch/C#/FundService/FundService/WithPattern.cs
using FundService.Common;
using System;
using System.Collections.Generic;
using static FundService.DomainModel;
using static FundService.PatchInstructionService;
using static FundService.Common.Pipeline;

namespace FundService
{
    public static class WithPattern {
        public static void PatchFund(int id, IEnumerable<PatchFundDto> instructions, Func<int, Fund> getFund, Func<Fund, Result<Fund>> updateFund) {
            var fund = getFund(id);
            var funcs = new List<Func<Fund, Result<Fund>>>();

            instructions.ToList().ForEach(instruction => {
                switch (instruction.Path) {
                    case FUND_TYPE:
                        funcs.Add(PatchFundType(fund, instruction));
                        break;
                    case FUND_NAME:
                        funcs.Add(PatchFundName(fund, instruction));
                        break;
                }
            });

            funcs.Add(ValidateFundName);
            funcs.Add(ValidateFundType);

            var result = RunUntilFirstFault(fund, funcs);
            switch (result.Success) {
                case true:
                    updateFund(fund);
                    break;
                default:
                    throw new SupportException(result.Error);
            }
        }
    }
}

File: http patch/Scala/fundservice/fund/src/main/scala/fund/WithPattern.scala
package fund

import fund.DomainModel._
import fund.PatchInstructionService._
import fund.common.Pipeline._
import fund.common.SupportException

import scala.collection.mutable.ListBuffer

object WithPattern {

  def patchFind(id: Int, instructions: List[PatchFundDto], getFund: ((Int) => Fund), updateFund: ((Fund) => Result[Fund])): Unit = {
    val fund = getFund(id)
    val funcs = ListBuffer[((Fund) => Result[Fund])]()

    instructions.foreach(instruction => {
      instruction.path match {
        case "fundType" => funcs += patchFundType(fund, instruction)
        case "fundName" => funcs += patchFundType(fund, instruction)
      }
    })

    funcs += publishUpdate(fund, println)
    funcs += validateFundName
    funcs += validateFundType

    runUntilFirstFault(fund, funcs.toList) match {
      case result if result.success() => updateFund(result.data)
      case result => throw new SupportException(result.error)
    }
  }
}
